build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,266 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { MonoRepo, Package } from "@fluidframework/build-tools";
+import async from "async";
+import chalk from "chalk";
+import execa from "execa";
+import { ResetMode } from "simple-git";
+
+import { findPackageOrReleaseGroup, packageOrReleaseGroupArg } from "../../args.js";
+import {
+	BaseCommand,
+	type Context,
+	Repository,
+	getPreReleaseDependencies,
+} from "../../library/index.js";
+
+type CheckFunction = (
+	context: Context,
+	rgOrPkg: MonoRepo | Package,
+	print?: boolean,
+) => Promise<CheckResult>;
+
+interface CheckResult {
+	result: boolean;
+	message?: string;
+	fixCommand?: string;
+	fatal?: boolean;
+}
```

Comment:
Again no docs on the type or its members makes it not super clear whats going on.

I assume there are suppose to be 3 states this can return, passing, nonfatal error, and fatal error, but this type allows for 6 combinations of result and fatal. Maybe a single enum would be better than a boolean and an optional boolean? Or at least document them a but.


Also, if you are going to have an undocumented result boolean, it might be better to repalce it with something that better indicates what try and false for it would mean. Currently its not super clear if true means it an error was reported (as is common with non-zero error codes meaing errors) or if true emans it passed the check and is good. Renaming result to "passing" or replacing the boolean with something like "pass" | "fail" | "fatal-error" would clear that up.